### ISSUE TITLE
TextTrackList onchange not working in Microsoft Edge

### ISSUE THREAD

**Issue by wannabefro** [videojs/video.js#2438]
The onchange event is not being triggered on a TextTrackList in Microsoft Edge.
I've produced a simple [example bin here](https://jsbin.com/qanoru/edit?console,output)
Expected behaviour is when switching between textTracks, "changed track" should be logged to the console.

The output of what player.textTracks() is below

```javascript
[object TextTrackList] {
  0: [object TextTrack] {
    activeCues: null,
    addCue: function addCue() { [native code] },
    addEventListener: function addEventListener() { [native code] },
    cues: [object TextTrackCueList] { ... },
    DISABLED: 0,
    dispatchEvent: function dispatchEvent() { [native code] },
    ERROR: 3,
    HIDDEN: 1,
    inBandMetadataTrackDispatchType: "",
    kind: "subtitles",
    label: "English",
    language: "en",
    LOADED: 2,
    LOADING: 1,
    mode: "disabled",
    NONE: 0,
    oncuechange: null,
    onerror: null,
    onload: null,
    readyState: 0,
    removeCue: function removeCue() { [native code] },
    removeEventListener: function removeEventListener() { [native code] },
    SHOWING: 2
  },
  addEventListener: function addEventListener() { [native code] },
  dispatchEvent: function dispatchEvent() { [native code] },
  item: function item() { [native code] },
  length: 1,
  onaddtrack: null,
  removeEventListener: function removeEventListener() { [native code] }
}
```

I noticed that both onchange and onremovetrack are missing from it.

**Comment by heff** [comment]
I don't have access to a VM with Edge yet. Would you be able to try the 5.0 branch and see if that works any better?
http://jsbin.com/sofiwu/edit?html,console,output

**Comment by wannabefro** [comment]
I tried it [using rc39 here](https://jsbin.com/bememi/1/edit?html,js,output) and can't get subtitles to appear in the first place in edge. `player.textTracks().length` is 0.

Removing videojs in it's entirety (so just the video element) shows the subtitles correctly.

**Comment by mmcc** [comment]
\*Insert snarky Microsoft browser comment\*

That's helpful, I'll grab a copy of Edge on one my IE VMs and see if I figure out anything else useful.

**Comment by wannabefro** [comment]
thanks! :smiley:

If there's anything I can help with in the meantime just let me know. We've got a windows 10 machine set up in the office so it's pretty straight forward for me to test things in edge.

**Comment by mmcc** [comment]
Excellent, we'll definitely take you up on that.

**Comment by gkatsev** [comment]
I wonder if Edge has issues with unclosed `track` or `source` elements.
Also, can you please report what you get for `videojs.getComponent('Html5').supportsNativeTextTracks()` in Edge? We should narrow down whether this is an issue with native tracks or not.

**Comment by wannabefro** [comment]
` videojs.getComponent('Html5').supportsNativeTextTracks() ` returns true in edge

**Comment by hartman** [comment]
PS poking the [ MS Edge dev team] (https://twitter.com/MSEdgeDev) on twitter really works at times and of course upstream issues when confirmed should really be reported at https://connect.microsoft.com/ie/feedback

**Comment by gkatsev** [comment]
Yep, looks like while it supports tracks, it doesn't support events on these tracks.
I filed a bug against IE11 (https://connect.microsoft.com/IE/feedbackdetail/view/1660701/text-tracks-do-not-fire-change-addtrack-or-removetrack-events) and tweeted at MSEdge to ask if there's someplace better to file a bug for it (https://twitter.com/gkatsev/status/631571823155638272)

**Comment by gkatsev** [comment]
To fix #2465 we turned off native tracks on IE11 (and there by Edge) in #2466. And this there's a bug filed against IE/Edge, I'll close this. I'll update it if as necessary when there are updates on the bug.
